internal/support/cronscheduler/application/scheduleactionrecord.go

```diff
@@ -156,14 +158,14 @@ func GenerateMissedScheduleActionRecords(ctx context.Context, dic *di.Container,
 
 			if _, err := dbClient.AddScheduleActionRecords(ctx, missedRecords); err != nil {
 				lc.Errorf("Failed to add missed schedule action records with action id: %s of job: %s to database. Correlation-ID: %s", actionId, job.Name, correlationId)
-				return errors.NewCommonEdgeXWrapper(err)
+				return errors.NewCommonEdgeXWrapper(err), hasMissedAction
 			}
 
 			lc.Debugf("Missed schedule action records with action id: %s of job: %s have been created successfully. Correlation-ID: %s", actionId, job.Name, correlationId)
 		}
 	}
 
-	return nil
+	return nil, hasMissedAction
```

Comment:
```suggestion
	return nil, len(missedRecords) > 0
```
declaire `missedRecords` in the beginning

Reply:
use this statement for all the return
try to avoid the individual boolean flag if possible